cmake/OpenCVCompilerOptions.cmake

```diff
@@ -146,8 +146,11 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   elseif(X86 OR X86_64)
     add_extra_compiler_option(-mno-sse2)
   endif()
+  if(ARM)
+    add_extra_compiler_option("-mfp16-format=ieee")
+  endif(ARM)
   if(ENABLE_NEON)
-    add_extra_compiler_option("-mfpu=neon")
+    add_extra_compiler_option("-mfpu=neon-fp16")
```

Comment:
what's the status of neon-fp16 support? Does any ARMv7 chip support it? What about compiler support? gcc/clang?